sbncode/SBNEventWeight/App/SystToolsEventWeight_module.cc

```diff
@@ -191,25 +204,92 @@ void SystToolsEventWeight::beginRun(art::Run& run) {
                                   << "sp->GetFullyQualifiedName() = " << sp->GetFullyQualifiedName() << "\n"
                                   << "sp->GetInstanceName() = " << sp->GetInstanceName() << "\n"
                                   << "Printing each SystParamHeader of this ISystProviderTool..";
-    //==== Note: typedef std::vector<SystParamHeader> SystMetaData;
+    // Note: typedef std::vector<SystParamHeader> SystMetaData;
     auto const& smd = sp->GetSystMetaData();
+
+    // make a map of responsless-response params
+    std::map<systtools::paramId_t, std::vector<systtools::paramId_t>> map_resp_to_respless;
     for( auto &sph : smd ){
-      if (fDebugMode) {
-        std::cout << "  sph.prettyName = " << sph.prettyName << std::endl;
+      if(sph.isResponselessParam){
+        auto it = map_resp_to_respless.find( sph.responseParamId );
+        if( it != map_resp_to_respless.end() ){
+          it->second.push_back( sph.systParamId );
+        }
+        else{
+          map_resp_to_respless[sph.responseParamId] = {};
+          map_resp_to_respless[sph.responseParamId].push_back( sph.systParamId );
+        }
       }
+    }
+    if (fDebugMode) {
+      for(const auto& it: map_resp_to_respless){
+        const auto& sph = systtools::GetParam(smd, it.first);
+        std::cout << "Found a dependent dial: " << sph.prettyName << std::endl;
+        for(const auto& depdialid: it.second){
+          const auto& sph_dep = systtools::GetParam(smd, depdialid);
+          std::cout << "- dep dial: " << sph_dep.prettyName << std::endl;
+        }
+      }
+    }
 
-      std::string rwmode = "multisigma";
-      if(sph.isRandomlyThrown) rwmode = "multisim";
+    for( auto &sph : smd ){
 
-      if (fDebugMode) {
-        std::cout << "  rwmode = " << rwmode << std::endl;
+      // responsless
+      if(sph.isResponselessParam){
+        if (fDebugMode) {
+          std::cout << "Responsless dial found: " << sph.prettyName << ", thus skipping" << std::endl;
+        }
+        continue;
       }
 
-      std::vector<float> widths { sph.paramVariations.begin(), sph.paramVariations.end() };
-
       sbn::evwgh::EventWeightParameterSet fParameterSet;
-      fParameterSet.AddParameter(sph.prettyName, std::move(widths));
-      fParameterSet.Configure(sp->GetFullyQualifiedName()+"_"+sph.prettyName, rwmode, sph.paramVariations.size());
+      std::string rwmode = "";
+
+      std::vector<double> paramVars = sph.isCorrection ? std::vector<double>(1, sph.centralParamValue) : sph.paramVariations;
```

Comment:
I think this is the same as:
```suggestion
      std::vector<double> paramVars = sph.isCorrection ? std::vector{ sph.centralParamValue } : sph.paramVariations;
```